apps/console/src/features/identity-providers/components/authenticator-grid.tsx

```diff
@@ -192,26 +205,27 @@ export const AuthenticatorGrid: FunctionComponent<AuthenticatorGridPropsInterfac
                     });
 
                     const results: ApplicationBasicInterface[] = await Promise.all(
-                        appRequests.map(response => response.catch(error => {
-                            dispatch(addAlert({
-                                description: error?.description
-                                    || "Error occurred while trying to retrieve connected applications.",
-                                level: AlertLevels.ERROR,
-                                message: error?.message || "Error Occurred."
-                            }));
-                        }))
+                        appRequests.map((response: Promise<any>) => response
+                            .catch((error: AlertState) => {
+                                dispatch(addAlert({
+                                    description: error?.description
+                                        || "Error occurred while trying to retrieve connected applications.",
+                                    level: AlertLevels.ERROR,
+                                    message: error?.message || "Error Occurred."
+                                }));
+                            }))
                     );
 
                     const appNames: string[] = [];
 
-                    results.forEach((app) => {
+                    results.forEach((app: ApplicationBasicInterface) => {
                         appNames.push(app.name);
                     });
 
                     setConnectedApps(appNames);
                 }
             })
-            .catch((error) => {
+            .catch((error: AlertState) => {
```

Comment:
The type of the error should come from the `getIDPConnectedApps`. It should be either of type `AxiosError` or `IdentityApiException` if thrown.

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/3653/commits/4d9d7327a9f3509da0b1bc4492c8777dfdbec431 and https://github.com/wso2/identity-apps/pull/3653/commits/9b30a0d3fe7dd18a817cbc326811440d72e39f7b